src/dashboard/src/components/administration/views.py

```diff
@@ -223,6 +229,160 @@ def storage(request):
     system_directory_description = 'Available storage'
     return render(request, 'administration/locations.html', locals())
 
+def usage(request):
+    usage_dirs = _usage_dirs()
+
+    return render(request, 'administration/usage.html', locals())
+
+def _usage_dirs(calculate_usage=True):
+    # Put spaces before directories contained by the spaces
+    #
+    # Description is optional, but either a path or a location purpose (used to
+    # look up the path) should be specified
+    #
+    # If only certain sudirectories within a path should be deleted, set
+    # 'subdirectories' to a list of them
+    dir_defs = (
+        ('shared', {
+            'path': _usage_shared_dir_path(),
+        }),
+        ('dips', {
+            'description': 'DIP uploads',
+            'path': os.path.join('watchedDirectories', 'uploadedDIPs'),
+            'contained_by': 'shared'
+        }),
+        ('rejected', {
+            'description': 'Rejected',
+            'path': 'rejected',
+            'contained_by': 'shared'
+        }),
+        ('failed', {
+            'description': 'Failed',
+            'path': 'failed',
+            'contained_by': 'shared'
+        }),
+        ('tmp', {
+            'description': 'Temporary file storage',
+            'path': 'tmp',
+            'contained_by': 'shared'
+        })
+    )
+
+    dirs = collections.OrderedDict(dir_defs)
+
+    # Resolve location paths and make relative paths absolute
+    for directory in dirs:
+        if dirs[directory].get('location_purpose') is not None:
+            # If a location, determine path
+            dirs[directory]['path'] = _usage_location_path(dirs[directory]['location_purpose'])
+        elif dirs[directory].get('contained_by') is not None:
+            # If contained, make path absolute
+            space = dirs[directory]['contained_by']
+            absolute_path = os.path.join(dirs[space]['path'], dirs[directory]['path'])
+            dirs[directory]['path'] = absolute_path
```

Comment:
More stylistic, but why do this in a loop, instead of in the definition? The shared directory path should be the same for all of them, so another option is to define a shared_dir variable, and os.path.join it to everything.